src/App.tsx

```diff
@@ -1,60 +1,118 @@
-import classNames from 'classnames';
+import { FC, useEffect, useState } from 'react';
+import cn from 'classnames';
 
-import 'bulma/css/bulma.css';
-import '@fortawesome/fontawesome-free/css/all.css';
-import './App.scss';
+import { User } from './types/User';
+import { Post } from './types/Post';
+
+import { getPosts, getUsers } from './utils/api';
 
 import { PostsList } from './components/PostsList';
 import { PostDetails } from './components/PostDetails';
 import { UserSelector } from './components/UserSelector';
 import { Loader } from './components/Loader';
 
-export const App = () => (
-  <main className="section">
-    <div className="container">
-      <div className="tile is-ancestor">
-        <div className="tile is-parent">
-          <div className="tile is-child box is-success">
-            <div className="block">
-              <UserSelector />
-            </div>
+import 'bulma/css/bulma.css';
+import '@fortawesome/fontawesome-free/css/all.css';
+import './App.scss';
 
-            <div className="block" data-cy="MainContent">
-              <p data-cy="NoSelectedUser">No user selected</p>
+export const App: FC = () => {
+  const [users, setUsers] = useState<User[]>([]);
+  const [selectedUser, setSelectedUser] = useState<User | null>(null);
 
-              <Loader />
+  const [userPosts, setUserPosts] = useState<Post[]>([]);
+  const [selectedPost, setSelectedPost] = useState<Post | null>(null);
 
-              <div
-                className="notification is-danger"
-                data-cy="PostsLoadingError"
-              >
-                Something went wrong!
-              </div>
+  const [isLoading, setIsLoading] = useState(false);
+  const [isError, setIsError] = useState(false);
+
+  const noUsersPosts =
+    !userPosts.length && selectedUser && !isLoading && !isError;
+
+  useEffect(() => {
+    setIsLoading(true);
+    setIsError(false);
 
-              <div className="notification is-warning" data-cy="NoPostsYet">
-                No posts yet
+    getUsers()
+      .then(setUsers)
+      .catch(() => setIsError(true))
+      .finally(() => setIsLoading(false));
+  }, []);
+
+  useEffect(() => {
+    if (!selectedUser) {
+      return;
+    }
+
+    setIsLoading(true);
+    setIsError(false);
+    setUserPosts([]);
+    setSelectedPost(null);
+
+    getPosts(selectedUser.id)
+      .then(setUserPosts)
+      .catch(() => setIsError(true))
+      .finally(() => setIsLoading(false));
+  }, [selectedUser]);
```

Comment:
Similar to the previous useEffect, using a try-catch block for error handling is recommended. This aligns with the checklist's suggestion to use try-catch for error handling.